Adding limitation on maximum number of commits considered (#27040)
DataDog/documentation: content/en/dora_metrics/setup/deployments.md

- Change lead time stage breakdown metrics are only available for GitHub.
- Change lead time is not available for the first deployment of a service that includes Git information.

> - The maximum number of commits considered when computing Change Lead Time for a specific deployment is 5000.

Reviewer: ```suggestion
- The Change Lead Time calculation includes a maximum of 5000 commits per deployment.
```
